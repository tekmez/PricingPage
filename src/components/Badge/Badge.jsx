import React from 'react'
import { View, Text } from 'react-native'
import styles from './Badge.style'
import YoutubeIcon from '../../../assets/svg/youtubeIcon.svg'
const Badge = () => {
  return (
    <View style={styles.container}>
      <YoutubeIcon width={20} height={20} />
      <Text style={styles.text}>Youtube Premium</Text>
    </View>
  )
}

export default Badge
