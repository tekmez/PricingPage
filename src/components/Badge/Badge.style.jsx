import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFF5F5',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
    paddingVertical: 6,
    paddingHorizontal: 12,
    borderRadius: 6,
    width: 174,
    height: 32,
  },
  text: {
    color: '#FF0000',
    fontFamily: 'Inter-Bold',
    fontSize: 14,
    lineHeight: 18,
  },
})
export default styles
