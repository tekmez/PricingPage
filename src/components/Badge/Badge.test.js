import React from 'react'
import { render } from '@testing-library/react-native'
import Badge from './Badge'
import styles from './Badge.style'
jest.mock('../../../assets/svg/youtubeIcon.svg', () => () => (
  <div testID="youtube-icon" />
))

describe('Badge component', () => {
  it('should match snapshot', () => {
    const tree = render(<Badge />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const comp = render(<Badge />)
    const badgeTitle = comp.getByTestId('badge-title').children[0]
    expect(badgeTitle).toBe('Youtube Premium')
  })

  it('should render style', () => {
    const comp = render(<Badge />)
    const badgeStyle = comp.getByTestId('badge-container').props.style
    expect(badgeStyle).toMatchObject(styles.container)
  })
})
