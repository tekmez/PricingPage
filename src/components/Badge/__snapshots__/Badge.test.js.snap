// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF5F5",
      "borderRadius": 6,
      "flexDirection": "row",
      "gap": 8,
      "height": 32,
      "paddingHorizontal": 12,
      "paddingVertical": 6,
      "width": 174,
    }
  }
  testID="badge-container"
>
  <div
    testID="youtube-icon"
  />
  <Text
    style={
      {
        "color": "#FF0000",
        "fontFamily": "Inter-Bold",
        "fontSize": 14,
        "lineHeight": 18,
      }
    }
    testID="badge-title"
  >
    Youtube Premium
  </Text>
</View>
`;
